pkg/services/cleanup/cleanup.go

```diff
@@ -57,66 +62,97 @@ type CleanUpService struct {
 	tempUserService           tempuser.Service
 }
 
+type cleanUpJob struct {
+	name string
+	fn   func(context.Context, log.Logger)
+}
+
+func (j cleanUpJob) String() string {
+	return strconv.Quote(j.name)
+}
+
 func (srv *CleanUpService) Run(ctx context.Context) error {
-	srv.cleanUpTmpFiles()
+	srv.cleanUpTmpFiles(ctx, srv.log)
 
 	ticker := time.NewTicker(time.Minute * 10)
 	for {
 		select {
 		case <-ticker.C:
-			ctxWithTimeout, cancelFn := context.WithTimeout(ctx, time.Minute*9)
-			defer cancelFn()
-
-			srv.cleanUpTmpFiles()
-			srv.deleteExpiredSnapshots(ctx)
-			srv.deleteExpiredDashboardVersions(ctx)
-			srv.deleteExpiredImages(ctx)
-			srv.cleanUpOldAnnotations(ctxWithTimeout)
-			srv.expireOldUserInvites(ctx)
-			srv.deleteStaleShortURLs(ctx)
-			srv.deleteStaleQueryHistory(ctx)
-			err := srv.ServerLockService.LockAndExecute(ctx, "delete old login attempts",
-				time.Minute*10, func(context.Context) {
-					srv.deleteOldLoginAttempts(ctx)
-				})
-			if err != nil {
-				srv.log.Error("failed to lock and execute cleanup of old login attempts", "error", err)
-			}
+			srv.clean(ctx)
 		case <-ctx.Done():
 			return ctx.Err()
 		}
 	}
 }
 
-func (srv *CleanUpService) cleanUpOldAnnotations(ctx context.Context) {
+func (srv *CleanUpService) clean(ctx context.Context) {
+	const timeout = 9 * time.Minute
```

Comment:
What do you think about making this timeout configurable? Timeouts are generally high on the list of things to mess with during outages. (genuine question, I have no objection to this as it is)

Reply:
I was thinking about it, but I don't want to introduce those changes in this PR.

Reply:
I've passed the code a few times and thought about doing it but I've never seen an issue about it. perhaps leavening at as is until someone claims they need it?